### PR TITLE
AB#39199 Connect GE web app to devx api 

### PR DESCRIPTION
**Pull request by omersayshi**
## Overview

Connecting GE web app to devx api for requesting queries when in application mode.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* Checkout to this branch
* Install and run the app 
* Switch into application mode
* Make queries to a Teams Team or chat with this app (screenshot showed below) installed in Garage Graph Explorer's test tenant.

NOTE: Message me if you need help with the last step.

image

### CODE REVIEW

**Comment by ayubun** [comment]
I am slightly confused with the testing here; So basically, we need to run the teams app alongside, connect it to a resource, and then run team queries on it in this PR of web app? Or am I misunderstanding 👁️

**Review by lhao03 — APPROVED** [review]
lgtm \^o^/
it's a hard to tell what was changed because it seems like all the lines were deleted and then added back in (it's all green), but it works on my end. just the comment about null/undefined check

File: src/app/services/actions/query-action-creator-util.ts [hunk 1]
```diff
@@ -1,158 +1,200 @@
...
+
+    dispatch(queryRunningStatus(true));
+
+    const graphEndpoint = query.sampleUrl.replace(GRAPH_URL, '');
+    const requestUrl = `${devxapiUrl}/graphproxy${graphEndpoint}`;
+
+    const response = await fetch(requestUrl, options);
+    if (!response.ok) {
+      throw response;
+    }
```
Comment (lhao03) [inline src/app/services/actions/query-action-creator-util.ts]: Why are we throwing the response?

**Review by lhao03 — CHANGES_REQUESTED** [review]

File: src/app/services/actions/query-action-creator-util.ts (hunk 1 above)
Comment (omersayshi) [reply src/app/services/actions/query-action-creator-util.ts]: Oops, yes we dont need to throw the error here. Thank you

**Comment by omersayshi** [comment]
> I am slightly confused with the testing here; So basically, we need to run the teams app alongside, connect it to a resource, and then run team queries on it in this PR of web app? Or am I misunderstanding 👁️

@AyuAi Yes, you need to locally run the Teams App and have the app that I screenshotted installed in a Team or a chat. Then run a query using that team or chat's resource

**Review by lhao03 — APPROVED** [review]

**Comment by lhao03** [comment]
In this PR: https://github.com/LokiLabs/microsoft-graph-explorer-v4/pull/59/files I changed it to return the response (no longer throwing) as I use the response to generate the relevant error message

**Comment by gracet9182** [comment]
> lgtm ^o^/
> it's a hard to tell what was changed because it seems like all the lines were deleted and then added back in (it's all green), but it works on my end. just the comment about null/undefined check

I also think it's hard to review when the the same things are deleted and re-added 😅 . Isn't it more prone to error? But I tested it and I can see it! nice

**Review by gracet9182 — APPROVED** [review]

**Pull request merged**